src/routes/safe/components/Apps/components/SecurityFeedbackModal/Slider.tsx

```diff
@@ -0,0 +1,186 @@
+import React, { useState, useEffect, useMemo } from 'react'
+import styled from 'styled-components'
+import { Box } from '@material-ui/core'
+import { Button } from '@gnosis.pm/safe-react-components'
+
+type SliderProps = {
+  onCancel: () => void
+  onSlideChange: (slideIndex: number) => void
+  onComplete: () => void
+}
+
+type SliderState = {
+  translate: number
+  activeSlide: number
+  transition: number
+  renderedSlides: React.ReactElement[]
+}
+
+const SLIDER_TIMEOUT = 500
+
+const Slider: React.FC<SliderProps> = ({ onCancel, onSlideChange, onComplete, children }) => {
```

Comment:
This component is very complex. I think that we can remove all the `useState`,  `useEffect` and `useRef` here and just creating a `activeStep` prop  we can do the same logic: 

```suggestion
import React, { useState, useEffect, useMemo } from 'react'
import styled from 'styled-components'
import { Box } from '@material-ui/core'
import { Button } from '@gnosis.pm/safe-react-components'

type SliderProps = {
  onSlideChange: (slideIndex: number) => void
  activeStep: number
}

const SLIDER_TIMEOUT = 500

const Slider: React.FC<SliderProps> = ({ onSlideChange, activeStep, children }) => {
  const allSlides = useMemo(() => React.Children.toArray(children).filter(Boolean) as React.ReactElement[], [children])

  const [disabledBtn, setDisabledBtn] = useState(false)

  useEffect(() => {
    let id: ReturnType<typeof setTimeout>

    if (disabledBtn) {
      id = setTimeout(() => {
        setDisabledBtn(false)
      }, SLIDER_TIMEOUT)
    }

    return () => {
      if (id) clearTimeout(id)
    }
  }, [disabledBtn])

  const nextSlide = () => {
    if (disabledBtn) return

    const nextStep = activeStep + 1

    onSlideChange(nextStep)
    setDisabledBtn(true)
  }

  const prevSlide = () => {
    if (disabledBtn) return

    const prevStep = activeStep - 1

    onSlideChange(prevStep)
    setDisabledBtn(true)
  }

  const isFirstStep = activeStep === 1

  return (
    <>
      <StyledContainer className="container">
        <StyledInner
          className="inner"
          style={{
            transition: `transform 0.5s ease`,
            transform: `translateX(-${activeStep * 100}%)`,
          }}
        >
          {allSlides.map((slide, index) => (
            <SliderItem key={index}>{slide}</SliderItem>
          ))}
        </StyledInner>
      </StyledContainer>
      <Box display="flex" justifyContent="center" width="100%">
        <Button color="primary" size="md" variant="bordered" fullWidth onClick={prevSlide}>
          {isFirstStep ? 'Cancel' : 'Back'}
        </Button>

        <Button
          color="primary"
          size="md"
          fullWidth
          onClick={nextSlide}
          style={{
            marginLeft: 10,
          }}
        >
          Continue
        </Button>
      </Box>
    </>
  )
}

const StyledContainer = styled.div`
  position: relative;
  margin: 0 auto;
  width: 100%;
  height: 100%;
  overflow: hidden;
`

const StyledInner = styled.div`
  position: relative;
  display: flex;
  min-width: 100%;
  min-height: 100%;
  transform: translateX(0);
  height: 426px;
`

const SliderItem = styled.div`
  min-width: 100%;
  min-height: 100%;

  background-position: center;
  background-repeat: no-repeat;
  background-size: cover;
`

export default Slider

```